text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,426 @@
+# Meta
+[meta]: #meta
+- Name: Asset Cache
+- Start Date: 2020-04-27
+- Author(s): dwillist
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: NA
+
+# Summary
+[summary]: #summary
+
+This change allows buildpacks to access cached assets during builds to enable offline builds and asset reuse.
+
+To do so we introduce the asset cache, a distribution artifact that provides vendored assets for buildpacks, and platform mechanisms to provide vendored assets to buildpacks during `build` and `detect`.
+
+# Motivation
+[motivation]: #motivation
+
+- Simplify artifacts needed to achieve offline builds.
+- Enable registry deduplication of buildpack assets
+- Separate buildpacks from vendored assets.
+
+# What it is
+[what-it-is]: #what-it-is
+
+### Define the target persona:
+Buildpack user, Platform operator, Buildpack author
+
+### Explaining the feature largely in terms of examples.
+
+##### Asset Cache Definition
+We define the asset cache artifact. It is a reproducible OCI image. Each layer in this image contains one or more `asset` files a buildpack may contribute during a future build. Each of these `asset` files will be made available at `/cnb/assets/<asset-digest>` in the build container.
+
+Asset Cache Layers Layout:
+```
+<layer1> ┳━ /cnb/assets/<java11-asset-digest>
+         ┗━ /cnb/assets/<java13-asset-digest>
+
+...
+<layern> ━━ /cnb/assets/<java15-asset-digest>
+```
+
+##### Asset Image Labels
+Asset caches will have two labels. A simple `io.buildpacks.asset.metadata` label that contains the asset's name.
+``` json
+{
+  "name": "asset-org/asset-name",
+}
+
+```
+
+Asset caches will also have a `io.buildpacks.asset.layers` Label. This label's contents will be a `json` object mapping each layer to metadata describing the assets it provides. Note that only the `digest` field is required in the metadata mapping.
+
+``` json
+{
+  "<layer1-diffID>": [
+    {
+      "name": "java11",
+      "digest": "sha256:java11-asset-sha256",
+      "uri": "https://path/to/java11.tgz",
+    },
+    {
+      "digest": "sha256:<java13-asset-sha256>",
+      "metadata": {
+        "name": "java13"
+      }
+    }
+  ],
+  "...": [],
+  "<layern-diffID>": [
+    {
+      "digest": "sha256:<java15-asset-sha256>",
+    }
+  ]
+}
+```
+
+##### Buildpackage Changes
+
+To enable buildpackages to reference asset caches, we define a new buildpackage label, `io.buildpacks.buildpackage.assets`.
```

Comment:
Could this be named `io.buildpacks.asset.layers` to maintain the same name across buildpackages/builders/asset packages?